dts/arm/nxp/nxp_rt118x.dtsi

```diff
@@ -100,3 +220,29 @@
 		status = "disabled";
 		clocks = <&ccm IMX_CCM_FLEXSPI_CLK 0x0 0>;
 };
+
+&flexspi2 {
+	compatible = "nxp,imx-flexspi";
+	interrupts = <56 0>;
+	#address-cells = <1>;
+	#size-cells = <0>;
+	status = "disabled";
+	clocks = <&ccm IMX_CCM_FLEXSPI2_CLK 0x0 0>;
+};
+
```

Comment:
I don't think the changes above this point are related to the PR objective?

Reply:
Sorry, originally thought that together can save a PR review process.  I have created another PR to complement full LPUART and GPIO devices, and ocram spaces. 
https://github.com/zephyrproject-rtos/zephyr/pull/77258